### PR TITLE
Fix PSR container conflicts

### LINKED ISSUE #7488: Fatal error: Declaration of WCPay\Internal\DependencyManagement\DelegateContainer\LegacyContainer::has($id) must be compatible with Psr\Container\ContainerInterface::has(string $id): bool
With PHP 8.2.4 I got error

```
 Fatal error: Declaration of WCPay\Internal\DependencyManagement\DelegateContainer\LegacyContainer::has($id) must be compatible with Psr\Container\ContainerInterface::has(string $id): bool
```
In 3 files:
- woocommerce-payments/src/Container.php
- woocommerce-payments/src/Internal/DependencyManagement/DelegateContainer/LegacyContainer.php
- woocommerce-payments/src/Internal/DependencyManagement/DelegateContainer/WooContainer.php

### PR DESCRIPTION
**Pull request by RadoslavGeorgiev**
Fixes #7488

#### Changes proposed in this Pull Request

Fixes PSR container conflicts by moving `Psr\Container` within our namespace

#### Testing instructions

Overall, if the container package is in our namespace, and a simple checkout flow works, we should be good.

Specific Testing Instructions for #7488 
1. On current stable release, install and activate [backwpup](https://wordpress.org/plugins/backwpup/) plugin. Observe Payment dashboard screens are not accessible.
2. With this fix, check that Payment dashboards screens work, checkout and payment admin flows are functional

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7514` or branch name `fix/psr-container-conflicts` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/psr-container-conflicts) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3eca63da8648f05ca6584421a6c7fdc5565647bc
- Build time: 2023-10-19 11:25:28 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by naman03malhotra — COMMENTED** [review]
Some E2E tests are failing!

File: lib/composer.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
     "minimum-stability": "dev",
     "require-dev": {
         "league/container": "4.2",
-        "coenjacobs/mozart": "0.7.1"
+        "coenjacobs/mozart": "0.7.1",
+        "psr/container": "^1.1"
```
Comment (zmaglica) [inline lib/composer.json:8]: I would replace caret and use a fixed version instead, like `"psr/container": "1.1.2"`

**Review by zmaglica — APPROVED** [review]

File: lib/composer.json (hunk 1 above)
Comment (naman03malhotra) [reply lib/composer.json:8]: +1 to this

Comment (naman03malhotra) [reply lib/composer.json:8]: +1 to this

**Review by naman03malhotra — APPROVED** [review]
LGTM code wise but I have not tested it on JN site.

**Pull request merged**